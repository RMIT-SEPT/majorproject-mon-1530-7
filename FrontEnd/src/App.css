.App {
  text-align: center;
  background-color: rgba(164, 214, 197, 0.904);
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.page-container{
  display: flex;
  min-height: 100vh;
  flex-direction: column;
}

.content-wrap{
  flex:1;
}