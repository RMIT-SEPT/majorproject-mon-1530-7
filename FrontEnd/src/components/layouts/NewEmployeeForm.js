import React, {useState} from 'react'
import Card from 'react-bootstrap/Card';
import Select from 'react-select';

const NewEmployeeForm = () => {

    return (
        <Card className="shadow p-3 mb-5 bg-white rounded"  border="light" style={{ width: '68rem' }}>
            <Card.Body>

                <textarea class="form-control form-rounded text-center" placeholder="Name"></textarea>
                
                <br />
                
                <Select
                    isMulti
                    name="Availability"
                    placeholder="Availability"
                    options={[
                        { value: "Monday", label: "Monday" },
                        { value: "Tuesday", label: "Tuesday" },
                        { value: "Wednesday", label: "Wednesday" },
                        { value: "Thursday", label: "Thursday" },
                        { value: "Friday", label: "Friday" },
                        { value: "Saturday", label: "Saturday" },
                        { value: "Sunday", label: "Sunday" }
                      ]}
                    className="basic-multi-select"
                    classNamePrefix="select"
                />
            </Card.Body>
        </Card>
    );
}

export default NewEmployeeForm;