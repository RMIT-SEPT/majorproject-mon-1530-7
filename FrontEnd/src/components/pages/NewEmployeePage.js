import React, {Component} from 'react'
import NewEmployeeForm from '../layouts/NewEmployeeForm';
import { Container, Jumbotron, CardDeck, Form, Button} from 'react-bootstrap';

class NewEmployeePage extends Component {
NewEmployeePage
    constructor(props){

        super(props);
        this.state = {
            
            showError:false,
            showSuccess:false
        };

        this.handleSubmit = this.handleSubmit.bind(this); 
    }

    handleSubmit(event){

        if(this.state.showError){
            alert('Invalid Employee Creation: Please input a name and set availability.');
            event.preventDefault();
        }
        else{
            alert('Employee Successfully Added!');
            event.preventDefault();
        }
        
    }

    showSuccessAlert = () =>{

        this.setState({showSuccess:true});
    }

    hideSuccessAlert = () =>{

        this.setState({showSuccess:false})
    }

    showErrorAlert = () =>{

        this.setState({showError:true});
    }

    hideErrorAlert = () =>{

        this.setState({showError:false})
    }

    render() {

        return (
            <Form onSubmit={this.handleSubmit}>
            <Jumbotron id="jumbotron-cus-book-page">
                <h2 className="h2-cus-book-page" class="text-center">Add Employee</h2>
                <Container className="customerBookingPageContainer">
                    <CardDeck>
                        <NewEmployeeForm/>
                    </CardDeck>
                </Container>
                <Container className="makeCusBooking">
                    <Button variant="primary" size="lg" type="submit"
                        onClick={this.showErrorAlert}>
                        Add Employee!
                    </Button>
                </Container>
            </Jumbotron>
            </Form>
        )
    }
}

export default NewEmployeePage;
