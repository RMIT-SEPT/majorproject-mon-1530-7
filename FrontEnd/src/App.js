import React from "react";
import { Redirect } from 'react-router';
import NavigationBar from "./components/layouts/Navbar.js";
import Footer from "./components/layouts/Footer.js";
import { Switch, Route, BrowserRouter } from "react-router-dom";
import LoginPage from "./components/pages/LoginPage";
import HomePage from "./components/pages/HomePage";
import AboutPage from "./components/pages/AboutPage";
import ContactPage from "./components/pages/ContactPage";
import CustomerBookingPage from "./components/pages/CustomerBookingPage.js";
import NewEmployeePage from "./components/pages/NewEmployeePage.js";

import AdminPastBooking from "./components/pages/AdminPastBooking.js";
import CustomerPastBookingsPage from "./components/pages/CustomerPastBookingsPage.js";

import Dashboard from "./components/pages/Dashboard.js";
import AdminDashboard from "./components/pages/AdminDashboard.js";
import Account from "./components/pages/Account.js";
import Profile from "./components/pages/Profile.js";

import ManageEmp from "./components/pages/ManageEmp.js";
import EmpDetails from "./components/pages/EmpDetails.js";

import UserProfile from "./UserProfile.js"
import AdminPastBookingsPage from "./components/pages/AdminPastBookingsPage.js";

function App() {
  console.log(UserProfile.getLoggedIn())
  if(UserProfile.getLoggedIn()) {
    return (
      <div className="page-container">
        <div className="content-wrap">
          <BrowserRouter>
            <NavigationBar loggedIn={UserProfile.getToken()} />
            <Switch>
                <Route exact path="/" component={HomePage} />,
                <Redirect from="/login" to="dashboard" />,
                <Route path="/about" component={AboutPage} />,
                <Route path="/contact" component={ContactPage} />,
                <Route path="/booking" component={CustomerBookingPage} />,
                <Route
                  path="/customer-past-bookings"
                  component={CustomerPastBookingsPage}
                />,
                <Route path="/dashboard" component={Dashboard} >
                  {UserProfile.getAdmin() ? <Redirect to="/admin-dashboard" /> : <Dashboard />}
                </Route>,
                <Route path="/admin-dashboard" component={AdminDashboard} >
                  {UserProfile.getAdmin() ?   <AdminDashboard /> : <Redirect to="/dashboard" /> }
                </Route>,
                <Route path="/account" component={Account} >
                  {UserProfile.getLoggedIn() ? <Account /> : <Redirect to="/login"/>}
                </Route>,
                <Route path="/new-employee" component={NewEmployeePage} />,
                <Route path="/profile" component={Profile} />,
                <Route path="/manage-emp" component={ManageEmp} />,
                <Route path="/emp-details" component={EmpDetails} /> 
          
            </Switch>
          </BrowserRouter>
        </div>
        <Footer />
      </div>
    );
  }
  else {
    return (
      <div className="page-container">
        <div className="content-wrap">
          <BrowserRouter>
            <NavigationBar loggedIn={UserProfile.getToken()} />
            <Switch>
              
                <Route exact path="/" component={HomePage} />,
                <Route path="/login" component={LoginPage} />,
                <Route path="/about" component={AboutPage} />,
                <Route path="/contact" component={ContactPage} />,
                <Route path="/booking" component={CustomerBookingPage} />,
                <Route
                  path="/customer-past-bookings"
                  component={CustomerPastBookingsPage}
                />,
                <Redirect from="/dashboard" to="/login" />,
                <Route path="/admin-dashboard" component={AdminDashboard} >
                  {UserProfile.getAdmin() ?   <AdminDashboard /> : <Redirect to="/dashboard" /> }
                </Route>,
                <Redirect from="/account" to="/login" />,
                <Route path="/new-employee" component={NewEmployeePage} />,
                <Route path="/profile" component={Profile} />,
                <Route path="/manage-emp" component={ManageEmp} />,
                <Route path="/emp-details" component={EmpDetails} />  
            </Switch>
          </BrowserRouter>
        </div>
        <Footer />
      </div>
    );
  }
}
 

export default App;
