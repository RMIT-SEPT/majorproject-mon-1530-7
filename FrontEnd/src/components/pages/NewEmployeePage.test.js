import React from 'react';
import {MemoryRouter} from 'react-router';
import NewEmployeePage from './NewEmployeePage';
import NewEmployeeForm from '../layouts/NewEmployeeForm';
import App from '../../App';
import {mount, shallow} from 'enzyme';
import toJSON from 'enzyme-to-json'
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('Testing NewEmployeePage route', () => {
    it('Ensures NewEmployeePage cotains the correct information from the coresponding route', () => {
      const component = mount( <MemoryRouter initialEntries={["/new-employee"]}>
          <App/>
       </MemoryRouter>
      );
      expect(component.contains(NewEmployeePage));
    })

    it('renders correctly', () => {
      const wrapper = shallow(<NewEmployeePage />)

      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    it('checks new employee form is present', () => {
      const wrapper = mount(<NewEmployeePage />);

      expect(wrapper.contains(NewEmployeeForm)).toEqual(true);
    });

    it('renders without crashing', () => {
      shallow(<NewEmployeePage />);
    });
  })

