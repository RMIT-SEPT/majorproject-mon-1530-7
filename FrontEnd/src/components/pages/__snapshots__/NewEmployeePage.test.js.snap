// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing HomePage route renders correctly 1`] = `
<Form
  inline={false}
  onSubmit={[Function]}
>
  <Jumbotron
    fluid={false}
    id="jumbotron-cus-book-page"
  >
    <h2
      className="h2-cus-book-page"
    >
      Add Employee
    </h2>
    <Container
      className="customerBookingPageContainer"
      fluid={false}
    >
      <CardDeck>
        <NewEmployeeForm />
      </CardDeck>
    </Container>
    <Container
      className="makeCusBooking"
      fluid={false}
    >
      <button
        className="btn-filled-extended"
        onClick={[Function]}
      >
        Add Employee
      </button>
    </Container>
  </Jumbotron>
</Form>
`;
